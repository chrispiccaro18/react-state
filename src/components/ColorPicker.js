import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from '../index.css';

export default class ColorPicker extends PureComponent {
  static propTypes = {
    colors: PropTypes.array.isRequired
  }

  state = {
    divColor: 'white'
  }

  clickHandler = (event) => {
    this.setState({ divColor: event.target.textContent });
  }

  render() {
    const colorButtons = this.props.colors.map(color => {
      return <button key={color} onClick={this.clickHandler} className={styles[color]}>{color.toUpperCase()}</button>;
    });
    return (
      <>
        {colorButtons}
        <div className={this.state.divColor}></div>
      </>
    );
  }
}
