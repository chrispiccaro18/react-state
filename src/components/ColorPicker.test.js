import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  it('renders a ColorPicker', () => {
    const wrapper = shallow(<ColorPicker colors={['red', 'yellow', 'blue']}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
