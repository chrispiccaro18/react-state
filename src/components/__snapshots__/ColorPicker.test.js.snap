// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders a ColorPicker 1`] = `
<Fragment>
  <button
    className="red"
    key="red"
    onClick={[Function]}
  >
    RED
  </button>
  <button
    className="yellow"
    key="yellow"
    onClick={[Function]}
  >
    YELLOW
  </button>
  <button
    className="blue"
    key="blue"
    onClick={[Function]}
  >
    BLUE
  </button>
  <ColorDisplay
    color="white"
  />
</Fragment>
`;
